### PR TITLE
[LLD][COFF] Add Support for ARM64EC Import Thunks

### PR DESCRIPTION
**Pull request by cjacek**
ARM64EC import thunks function similarly to regular ARM64 thunks but use a mangled name and perform the call through the auxiliary IAT.

### CODE REVIEW

**Comment by cjacek** [comment]
Depends on  #108459.

File: lld/COFF/Symbols.cpp [hunk 1]
```diff
@@ -19,6 +19,7 @@
 
 using namespace llvm;
 using namespace llvm::object;
+using namespace llvm::COFF;
```
Comment (mstorsjo) [inline lld/COFF/Symbols.cpp:22]: This looks like a stray change?

File: lld/test/COFF/arm64ec-import.test [hunk 2]
```diff
@@ -39,33 +39,40 @@ RUN: llvm-objdump -d out2.dll | FileCheck --check-prefix=DISASM %s
...
-DISASM-NEXT: 180001038: 9000000a     adrp    x10, 0x180001000 <.text>
-DISASM-NEXT: 18000103c: 9101314a     add     x10, x10, #0x4c
-DISASM-NEXT: 180001040: 17fffff0     b       0x180001000 <.text>
-DISASM-NEXT: 180001044: 52800020     mov     w0, #0x1                // =1
-DISASM-NEXT: 180001048: d65f03c0     ret
-DISASM-NEXT: 18000104c: 52800040     mov     w0, #0x2                // =2
-DISASM-NEXT: 180001050: d65f03c0     ret
+DISASM-NEXT: 180001008: 90000030     adrp    x16, 0x180005000
+DISASM-NEXT: 18000100c: f9400610     ldr     x16, [x16, #0x8]
+DISASM-NEXT: 180001010: d61f0200     br      x16
```
Comment (mstorsjo) [inline lld/test/COFF/arm64ec-import.test:44]: So the existing thunks use x11/x10, but these ones uses x16 like they used to, for regular arm64. I had to check, but x16 does indeed seem to be kosher to use on arm64ec.

**Review by mstorsjo — APPROVED** [review]
LGTM

File: lld/COFF/Symbols.cpp (hunk 1 above)
Comment (cjacek) [reply lld/COFF/Symbols.cpp]: Yes. I removed that and merged, thanks!

**Pull request merged**